commercetools-sdk-base/src/main/java/io/sphere/sdk/client/SphereApiConfig.java

```diff
@@ -10,11 +10,17 @@
 
     String getProjectKey();
 
+    CorrelationIdGenerator getCorrelationIdGenerator();
```

Comment:
I would not expose this method in SphereApiConfig, you would break a lot of my projects by additionally requiring to implement this method which the user may not care about.
If you disagree to expose this, at least implement it as default method so you don't break the clients.